### PR TITLE
Gesso config update

### PR DESCRIPTION
**Pull request by sbrack8t**

### CODE REVIEW

File: _starter-kit/_patterns/00-global/01-util/_utill.convert.scss [hunk 1]
```diff
@@ -0,0 +1,11 @@
+/// Convert unitless number to unit value
```
Comment (coreylafferty) [inline _starter-kit/_patterns/00-global/01-util/_utill.convert.scss]: shouldn't filename be _util.convert.scss?

File: _starter-kit/_patterns/00-global/02-tools/_tools.link.scss [hunk 2]
```diff
@@ -0,0 +1,7 @@
+@mixin link($color: $color-text-link, $hover-color: $color-text-link-hover) {
+  color: $color;
+
+  &:hover {
```
Comment (coreylafferty) [inline _starter-kit/_patterns/00-global/02-tools/_tools.link.scss]: Shouldn't we have :focus here as well?

File: _starter-kit/_patterns/01-base/03-html-elements/01-structural-elements/04-article/_article.scss [hunk 3]
```diff
@@ -1,4 +1,4 @@
 // @file
 // Article element styles.
 
-article {}
+// article {}
```
Comment (coreylafferty) [inline _starter-kit/_patterns/01-base/03-html-elements/01-structural-elements/04-article/_article.scss:4]: Why comment all these empty tags out?  Sass doesn't compile them if theyr'e empty.

File: _starter-kit/_patterns/03-components/article/article.twig [hunk 4]
```diff
@@ -1,25 +1,18 @@
...
-%}
-
-<article{{- attributes.addClass(classes)|raw -}}>
+{% set classes = classes | merge(['article']) | join(' ') | trim %}
+<article{{-attributes.addClass(classes)|raw-}}>
   {% block title %}
-    {% include '@components/page-title/page-title.twig' with {
-      'page_title': article_page_title
-    } %}
+    {{ title|raw }}
```
Comment (coreylafferty) [inline _starter-kit/_patterns/03-components/article/article.twig]: Why'd you remove the call to the page-title component?  Now this will just print out plain text with no wrapper element.

File: _starter-kit/_patterns/03-components/button/_button.scss [hunk 5]
```diff
@@ -2,16 +2,17 @@
 // Styles for Button.
 
 .button {
+  @extend %button-font;
```
Comment (coreylafferty) [inline _starter-kit/_patterns/03-components/button/_button.scss:5]: Why an additional line for extending the button font?  For simplicity/consistency, shouldn't the button mixin take care of this?

File: _starter-kit/_patterns/03-components/details/_details.scss [hunk 6]
```diff
@@ -5,14 +5,18 @@ $details-box-shadow-focus: 0 0 3px #3e94cf, 0 0 7px #3e94cf !default;
...
 // 'details' can appear as a modernizr class on the html tag, so this
 // class is limited to only the details element
+// Fix of modernizr class issue
+// stylelint-disable-next-line selector-no-qualifying-type
 details.details {
-  margin: 0 0 rem($vertical-spacing-forms);
+  margin: 0 0 rem($spacing-scale-level-4);
 
   // Fallback for browsers that don’t support details.
+  // Fix of modernizr class issue
```
Comment (coreylafferty) [inline _starter-kit/_patterns/03-components/details/_details.scss:18]: I don't think this comment is accurate.

File: _starter-kit/_patterns/03-components/dropbutton/_dropbutton.scss [hunk 7]
```diff
@@ -37,7 +37,7 @@ $dropbutton-toggle-icon-size: rem(12px) !default;
   }
 
   & + &,
-  & + .button,
+  // stylelint-disable-next-line
```
Comment (coreylafferty) [inline _starter-kit/_patterns/03-components/dropbutton/_dropbutton.scss:40]: Why is this needed?

File: _starter-kit/_patterns/03-components/fieldset/_fieldset.scss [hunk 8]
```diff
@@ -1,12 +1,17 @@
 // @file
 // Styles for form fieldsets.
 
+$base-line-height: 1.5;
```
Comment (coreylafferty) [inline _starter-kit/_patterns/03-components/fieldset/_fieldset.scss:4]: Should probably rename these two variables if they are now specific to this component.

File: _starter-kit/_patterns/03-components/fieldset/_fieldset.scss [hunk 9]
```diff
@@ -51,13 +57,14 @@ $fieldset-legend-offset:  rem($base-line-height * $base-font-size) !default;
 .fieldset--default {
   background: #fff;
   border: 1px solid $fieldset-color-border;
-  margin: rem($vertical-spacing) 0 (rem($vertical-spacing) + rem(12px));
+  // TODO Fix this
```
Comment (coreylafferty) [inline _starter-kit/_patterns/03-components/fieldset/_fieldset.scss:60]: giterdun

File: _starter-kit/_patterns/03-components/form-item/_form-item.scss [hunk 10]
```diff
@@ -1,6 +1,37 @@
 // @file
 // Styles for form items.
 
+// TODO: remove all unused variables
```
Comment (coreylafferty) [inline _starter-kit/_patterns/03-components/form-item/_form-item.scss]: same here

File: _starter-kit/_patterns/03-components/menu/menu--mobile/_menu--mobile.scss [hunk 11]
```diff
@@ -56,9 +53,10 @@ $menu-mobile-submenu-hover-color: #000 !default;
   }
 
   .menu__subnav-arrow {
-    background: none;
     @include image-replace(100%, 44px);
-    @include svg-background-inline(mobile-arrow-down);
+    // @include svg-background-inline(mobile-arrow-down);
```
Comment (coreylafferty) [inline _starter-kit/_patterns/03-components/menu/menu--mobile/_menu--mobile.scss:57]: ?

File: _starter-kit/_patterns/_utility-classes.scss [hunk 12]
```diff
@@ -1,37 +1,89 @@
+// Spacing directions
```
Comment (coreylafferty) [inline _starter-kit/_patterns/_utility-classes.scss:1]: Shouldn't this top bit be in tools somewhere?

File: _starter-kit/_patterns/00-global/01-util/_utill.convert.scss (hunk 1 above)
Comment (sbrack8t) [reply _starter-kit/_patterns/00-global/01-util/_utill.convert.scss]: yup. typo. noted

File: _starter-kit/_patterns/03-components/article/article.twig (hunk 4 above)
Comment (sbrack8t) [reply _starter-kit/_patterns/03-components/article/article.twig]: This might be a copy paste error. Will revert.

File: _starter-kit/_patterns/00-global/01-util/_utill.convert.scss (hunk 1 above)
Comment (sbrack8t) [reply _starter-kit/_patterns/00-global/01-util/_utill.convert.scss]: Resolved

File: _starter-kit/_patterns/00-global/02-tools/_tools.link.scss (hunk 2 above)
Comment (sbrack8t) [reply _starter-kit/_patterns/00-global/02-tools/_tools.link.scss]: Added visited and active states

File: _starter-kit/_patterns/03-components/button/_button.scss (hunk 5 above)
Comment (sbrack8t) [reply _starter-kit/_patterns/03-components/button/_button.scss:5]: The goal of the extends, in general, was to have the means to capture both the responsive changes(font-size and margins) and the contextual padding needs ( [element] + h2 { padding-top: x} in a reusable way. Mixins allow for the former but not the latter. In this instance, %button-font handles the type while the mixin manages the button appearance and interaction. Not wedded to the approach but was trying to stay consistent with managing type separately

File: _starter-kit/_patterns/03-components/fieldset/_fieldset.scss (hunk 8 above)
Comment (sbrack8t) [reply _starter-kit/_patterns/03-components/fieldset/_fieldset.scss:4]: The field and fieldset files contain a lot of self-contained variables fonts and mixins. I would like to address them in separate ticket once we resolve the config files and approach

File: _starter-kit/_patterns/03-components/fieldset/_fieldset.scss (hunk 9 above)
Comment (sbrack8t) [reply _starter-kit/_patterns/03-components/fieldset/_fieldset.scss:60]: See above comment

File: _starter-kit/_patterns/03-components/form-item/_form-item.scss (hunk 10 above)
Comment (sbrack8t) [reply _starter-kit/_patterns/03-components/form-item/_form-item.scss]: Removed

File: _starter-kit/_patterns/_utility-classes.scss (hunk 12 above)
Comment (sbrack8t) [reply _starter-kit/_patterns/_utility-classes.scss:1]: Yes and no.
